dotnet/package/common.csproj

```diff
@@ -38,83 +42,80 @@
-  <UsingTask TaskName="CreateFrameworkListFile" AssemblyFile="$(DotNetSharedFrameworkTaskFile)" Condition="'$(PackageType)' != 'Template'" />
+  <!-- Remove the KnownFrameworkReference declared in ../targets/Microsoft.$(_PlatformName).Sdk.Versions.props so that we do not attempt to restore/include the ref/runtime packs that we are trying to build -->
+  <Target Name="_RemoveThisFrameworkReference"
+      BeforeTargets="ProcessFrameworkReferences" >
+    <ItemGroup>
+      <KnownFrameworkReference Remove="Microsoft.$(_PlatformName)" />
+    </ItemGroup>
+  </Target>
 
-  <Target Name="_GenerateFrameworkListFile" Condition=" '$(_CreateFrameworkList)' == 'true' Or '$(_CreateRuntimeList)' == 'true' ">
-    <!-- https://github.com/dotnet/runtime/blob/0647ec314948904319da5eb15e9931f7c85ed1e2/src/installer/pkg/projects/Directory.Build.targets#L281 -->
-    <PropertyGroup Condition="'$(_CreateFrameworkList)' == 'true'">
-      <_FrameworkListFile>$(_packagePath)data/FrameworkList.xml</_FrameworkListFile>
+  <Target Name="_IncludePlatformPackageContent"
+      Condition=" '$(PlatformPackageType)' != '' "
+      BeforeTargets="GetFilesToPackage" >
+    <PropertyGroup Condition=" '$(PlatformPackageType)' == 'TargetingPack' ">
       <_PackTargetPath>ref/net$(BundledNETCoreAppTargetFrameworkVersion)</_PackTargetPath>
       <_PackNativePath>runtimes/$(_RuntimeIdentifier)/native</_PackNativePath>
     </PropertyGroup>
-    <PropertyGroup Condition="'$(_CreateRuntimeList)' == 'true'">
-      <_FrameworkListFile>$(_packagePath)data/RuntimeList.xml</_FrameworkListFile>
+    <PropertyGroup Condition=" '$(PlatformPackageType)' == 'RuntimePack' ">
       <_PackTargetPath>runtimes/$(_RuntimeIdentifier)/lib/net$(BundledNETCoreAppTargetFrameworkVersion)</_PackTargetPath>
       <_PackNativePath>runtimes/$(_RuntimeIdentifier)/native</_PackNativePath>
     </PropertyGroup>
-    <ItemGroup>
-      <_PackageFiles Include="$(_packagePath)$(_PackTargetPath)/*.dll" PackagePath="$(_PackTargetPath)" TargetPath="$(_PackTargetPath)" />
-      <_PackageFiles Include="$(_packagePath)$(_PackNativePath)/*.a" PackagePath="$(_PackNativePath)" TargetPath="$(_PackNativePath)" IsNative="true" />
-      <_PackageFiles Include="$(_packagePath)$(_PackNativePath)/*.dylib" PackagePath="$(_PackNativePath)" TargetPath="$(_PackNativePath)" IsNative="true" />
-      <_FrameworkListFileClass Include="@(_PackageFiles->'%(Filename)%(Extension)')" Profile="$(_PlatformName)" />
+    <ItemGroup Condition=" '$(PlatformPackageType)' == 'TargetingPack' or '$(PlatformPackageType)' == 'RuntimePack' " >
+      <_FxAssembliesToPackage Include="$(_packagePath)$(_PackTargetPath)/*.dll" />
+      <FilesToPackage Include="@(_FxAssembliesToPackage)" TargetPath="$(_PackTargetPath)" />
+      <FilesToPackage Include="@(_FxAssembliesToPackage->'%(RelativeDir)%(Filename).pdb')" TargetPath="$(_PackTargetPath)" IsSymbolFile="true" Condition=" '$(PlatformPackageType)' != 'RuntimePack' " />
```

Comment:
I'm a bit confused as to whether the runtime packs will still have the pdbs (further above a comment says "Always include symbols in runtime packs ..."), but I remembered one important detail: the pdb files are necessary during the build (and I don't think the build fetches symbols from *.symbols.nupkg), because the AOT compiler uses them to generate native debug information (dSYM bundles) - and because the native debug information bundles contain this information, the pdb files are not actually included in the app bundle.